### PR TITLE
Fix (most) clippy warnings

### PR DESCRIPTION
**Pull request by Neutroni**
Fixes most issues reported by `cargo clippy`
Most issues were about either explicit lifetimes given in parameter types where they could be elided or about 
using match for destructuring a single pattern.

Remaining two issues are about function having too many parameters and are not as easily fixed.

### CODE REVIEW

File: gnuplot/src/axes_common.rs [hunk 1]
```diff
@@ -1175,6 +1135,12 @@ pub struct Margins
 	pub bottom: Option<f32>,
 }
 
+impl Default for Margins {
```
Comment (SiegeLord) [inline gnuplot/src/axes_common.rs:1138]: Is this something clippy wanted? I don't see the value of this for a crate-private type if I don't use the `Default::default()`.

File: gnuplot/examples/inverse_api.rs [hunk 2]
```diff
@@ -77,14 +77,6 @@ fn lines<'l, Tx: IntoIterator<Item = f32>, Ty: IntoIterator<Item = f32>>(x: Tx,
 	}
 }
 
-impl Lines
```
Comment (SiegeLord) [inline gnuplot/examples/inverse_api.rs:80]: Please revert this. I guess it's not used, but the presense of this code is important to convey the idea of this API.

**Review by SiegeLord — CHANGES_REQUESTED** [review]
Thanks. Some of these changes are pretty suspect, but I don't mind merging them. Please run `cargo fmt` on this.

File: gnuplot/src/axes_common.rs (hunk 1 above)
Comment (Neutroni) [reply gnuplot/src/axes_common.rs:1138]: `warning: you should consider adding a `Default` implementation for `Margins`
    --> src/axes_common.rs:1147:2
     |
1147 | /     pub fn new() -> Self
1148 | |     {
1149 | |         Margins {
1150 | |             left: None,
...    |
1154 | |         }
1155 | |     }
     | |_____^
     |
     = note: `#[warn(clippy::new_without_default)]` on by default
     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#new_without_default
help: try adding this
     |
1145 + impl Default for Margins {
1146 +     fn default() -> Self {
1147 +         Self::new()
1148 +     }
1149 + }
`

File: gnuplot/examples/inverse_api.rs (hunk 2 above)
Comment (Neutroni) [reply gnuplot/examples/inverse_api.rs:80]: Reverted, added `#[allow(dead_code)]` to the Impl

**Comment by Neutroni** [comment]
Was the intent behind Rustfmt configuration merge_imports=false more similar to imports_granularity "Preserve" or "Item"?

**Review by SiegeLord — APPROVED** [review]

**Comment by SiegeLord** [comment]
"Preserve" is good enough for now.

**Pull request merged**